code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/dao/impl/PasswordDaoImpl.java

```diff
@@ -99,6 +119,39 @@ public Credential createAdminAccountAndPassword(String username, String password
         return objectManager.create(Credential.class, objectManager.convertToPropertiesFor(Credential.class, properties));
     }
 
+    public void validatePassword(String password) {
+        String authValidateUrl = AUTH_VALIDATE_URL.get();
+        if (!StringUtils.isBlank(authValidateUrl)) {
+            try {
+                Map<String, String> data = new HashMap<String, String>();
+                data.put("secret", password);
+                String jsonString = jsonMapper.writeValueAsString(data);
+                HttpResponse response =  Request.Post(authValidateUrl).bodyString(jsonString, ContentType.APPLICATION_JSON).execute().returnResponse();
+                if (response.getStatusLine().getStatusCode() >=400 && response.getStatusLine().getStatusCode() <= 499) {
+                    Map<String, Object> jsonData;
+
+                    jsonData = jsonMapper.readValue(response.getEntity().getContent());
+
+                    if (!jsonData.containsKey("type") || !jsonData.containsKey("message")) {
+                        throw new ClientVisibleException(ResponseCodes.INTERNAL_SERVER_ERROR, "Incomplete JSON response");
+                    }
+
+                    if (jsonData.get("type") != null) {
+                        throw new ClientVisibleException(ResponseCodes.UNPROCESSABLE_ENTITY, (String) jsonData.get("message"));
+                    }
+               }
+               else {
+                   if (response.getStatusLine().getStatusCode() < 200 || response.getStatusLine().getStatusCode() > 299) {
+                        throw new ClientVisibleException(ResponseCodes.INTERNAL_SERVER_ERROR, "Error talking to validator");
+                   }
+              }
+            } catch (IOException e) {
+                throw new ClientVisibleException(ResponseCodes.INTERNAL_SERVER_ERROR, "No response from validator");
+            }
+
+        }
+    }
+
```

Comment:
This should not be put in the Dao, may be added in a separate file say LocalAuthPasswordValidator in package io.cattle.platform.iaas.api.auth.integration.local